containers/flatpak/cockpit-client

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import subprocess
+import sys
+import gi
+
+gi.require_version("Gtk", "3.0")  # NOQA
+gi.require_version("WebKit2", "4.0")  # NOQA
+
+from gi.repository import Gtk, Gdk, WebKit2
+
+
+class Browser(Gtk.Window):
+    def __init__(self, uri):
+        super().__init__()
+        self.set_title("Web Console")
+
+        self.set_default_size(1200, 800)
+        self.set_size_request(800, 600)
+
+        context = WebKit2.WebContext.get_default()
+        data_manager = context.get_website_data_manager()
+        data_manager.set_network_proxy_settings(WebKit2.NetworkProxyMode.NO_PROXY, None)
```

Comment:
So I guess a flatpak runs as kinda-sorta-container in its own network namespace, but normally uses a network proxy to talk to the outside world?

Reply:
Yes and no.  You can add a `--share=network` to not unshare the network namespace, and then in that case, you're running as part of the host.

I decided not to do that: I'd prefer to shell out to ssh on the outside, to gain the advantages of the user's ssh config (aliases, existing keys, etc).

The shelling out to ssh is definitely the most problematic aspect of this entire thing, though.  It might be that we need to change just to use normal cockpit-ssh with libssh, and give the app access to the user's home directory.  That would require bundling libssh with the flatpak.  I didn't look into that.